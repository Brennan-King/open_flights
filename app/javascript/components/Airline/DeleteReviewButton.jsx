import React from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';
import './DeleteReviewButton.scss';

const propTypes = {
  /** The id of the review */
  reviewId: PropTypes.number,
  /** Callback function to set delete button state */
  setDeleteButtonClicked: PropTypes.func,
};

const propDefaults = {
  reviewId: 0,
  setDeleteButtonClicked: () => {},
};

/**
 * Constructs a delete button component used to delete a specific review.
 * @param {object} props the props used to configure the component
 * @return {node} a div containing the configured delete button
 */
const DeleteReviewButton = (props) => {
  const {reviewId, setDeleteButtonClicked} = props;

  const onDeleteReviewButtonClick = () => {
    const csrfToken = document.querySelector('[name=csrf-token]').content;
    axios.defaults.headers.common['X-CSRF-TOKEN'] = csrfToken;

    axios.delete(`/api/v1/reviews/${reviewId}`);
    setDeleteButtonClicked(true);
  };

  return (
    <div
      className="delete-review-button"
      onClick={() => onDeleteReviewButtonClick()}
    >
      <div className="x-symbol">x</div>
    </div>
  );
};

DeleteReviewButton.propTypes = propTypes;
DeleteReviewButton.propDefaults = propDefaults;
export default DeleteReviewButton;
