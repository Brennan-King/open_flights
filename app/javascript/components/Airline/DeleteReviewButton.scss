.delete-review-button {
  height: 23px;
  width: 18px;
  background-color: red;
  margin-left: auto;
  cursor: pointer;
  transition: ease-in-out 0.1s;
  border: 2px solid black;
  box-shadow: 2px 2px 2px grey;
}

.x-symbol {
  margin-top: -10px;
  color: white;
  font-weight: lighter;
}

.delete-review-button:hover {
  background: darkred;
}