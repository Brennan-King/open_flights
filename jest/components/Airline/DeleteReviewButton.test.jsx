/* eslint-disable max-len */
import 'jsdom-global/register';
import React from 'react';
import DeleteReviewButton from '../../../app/javascript/components/Airline/DeleteReviewButton';
import {configure, mount} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

configure({adapter: new Adapter()});

/**
 * Test suite for DeleteReviewButton component
 */
describe('DeleteReviewButton', () => {
  it('should render a DeleteReviewButton component', () => {
    const component = mount(
        <DeleteReviewButton
          reviewId={0}
          setDeleteButtonClicked={()=>{}}
        />,
    );

    expect(component).toMatchSnapshot();
  });

  it('should delete a review', () => {
    const mockSetDeleteButtonClicked = jest.fn();

    const component = mount(
        <DeleteReviewButton
          reviewId={0}
          setDeleteButtonClicked={mockSetDeleteButtonClicked}
        />,
    );

    component.find('#delete-review-button').simulate('click');
    expect(mockSetDeleteButtonClicked).toHaveBeenCalledTimes(1);
    expect(component).toMatchSnapshot();
  });
});
