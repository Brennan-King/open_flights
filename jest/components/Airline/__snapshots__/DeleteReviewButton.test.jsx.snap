// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteReviewButton should delete a review 1`] = `
<DeleteReviewButton
  reviewId={0}
  setDeleteButtonClicked={
    [MockFunction] {
      "calls": Array [
        Array [
          true,
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <div
    className="delete-review-button"
    id="delete-review-button"
    onClick={[Function]}
  >
    <div
      className="x-symbol"
    >
      x
    </div>
  </div>
</DeleteReviewButton>
`;

exports[`DeleteReviewButton should render a DeleteReviewButton component 1`] = `
<DeleteReviewButton
  reviewId={0}
  setDeleteButtonClicked={[Function]}
>
  <div
    className="delete-review-button"
    id="delete-review-button"
    onClick={[Function]}
  >
    <div
      className="x-symbol"
    >
      x
    </div>
  </div>
</DeleteReviewButton>
`;
